src/wombats/game/decisions/smoke.clj

```diff
@@ -0,0 +1,89 @@
+(ns wombats.game.decisions.smoke
+  (:require [wombats.game.decisions.helpers :as dh]
+            [wombats.game.utils :as gu]
+            [wombats.arena.utils :as au]))
+
+(defn- get-smoked-coords
+  "Return a collection of coords that should be smokescreened given a central starting point"
+  [coords arena-dims]
+  (conj (map #(gu/adjust-coords coords % arena-dims)
+             (range 8))
+        coords))
+
+(defn- calculate-throw-direction
+  "determines the direction smoke should be thrown based off of the
+  decision makers orientation and the specified throw direction"
+  [decision-maker-orientation throw-direction]
+  (condp = throw-direction
+    :forward decision-maker-orientation
+    :right (mod (+ decision-maker-orientation 2) 8)
+    :backward (mod (+ decision-maker-orientation 4) 8)
+    :left (mod (- decision-maker-orientation 2) 8)
+    nil))
+
+(defn- get-smoke-metadata
+  "determines the metadata to display in the arena"
+  [{{smoke-duration :arena/smoke-duration} :arena-config}
+   decision-maker-state]
+  (let [decision-maker (dh/get-decision-maker-contents decision-maker-state)]
+    {:type :smoke
+     :owner-id (:uuid decision-maker)
+     :color (get decision-maker :color "black")
+     :decay smoke-duration}))
```

Comment:
Will `:decay` be static then? It might be nice for it to show how much longer it will be present on the arena.